### ISSUE TITLE
Torch resnet inference [optimization_level_0-bfp8] fail

### ISSUE THREAD

**Issue by sdjukicTT** [tenstorrent/tt-xla#3163]
**Note:**
On new main error is:
```
Always | TT_FATAL: batch_norm only supports bfloat16, float32. Got: DataType::BFLOAT8_B
```

Repro:
```
pytest -svv tests/torch/models/resnet/test_resnet.py::test_torch_resnet_inference[optimization_level_0-bfp8]
```

### ...
never finishes executing if tt-mlir is on commit or newer:
Remove the generic ttir::PoolingOp and update SHLO-TTIR conversion pass (https://github.com/tenstorrent/tt-mlir/pull/6573)

Workflow runs that tested this:
https://github.com/tenstorrent/tt-xla/actions/runs/21684122324
https://github.com/tenstorrent/tt-xla/actions/runs/21684139211

**Comment by dgolubovicTT** [comment]
The issue was caused by `enable_bfp8_conversion` flag that casts whole graph to bfp8 (adds typecasts to all inputs/outputs to ensure whole graph is in bfp8).
 This is bad because many ops, including batch norm, don't support block formats. Therefore the error: "Always | TT_FATAL: batch_norm only supports bfloat16, float32. Got: DataType::BFLOAT8_B"
